battles/forms.py

```diff
@@ -24,29 +26,46 @@ class Meta:
         model = BattleTeam
         fields = []
 
-    first_pokemon = forms.IntegerField(
-        min_value=1, max_value=802, required=True, label='First pokemon',
-        validators=[validate_integer_is_not_zero])
-    second_pokemon = forms.IntegerField(
-        min_value=1, max_value=802, required=True, label='Second pokemon',
-        validators=[validate_integer_is_not_zero])
-    third_pokemon = forms.IntegerField(
-        min_value=1, max_value=802, required=True, label='Third pokemon',
-        validators=[validate_integer_is_not_zero])
+    first_pokemon = forms.ModelChoiceField(
+        queryset=Pokemon.objects.all(),
+        widget=autocomplete.ModelSelect2(
+            url='/battles/pokemon-autocomplete',
+            attrs={
+                'data-placeholder': 'Select the first Pokemon',
+                'data-html': True
+            })
+    )
+
+    second_pokemon = forms.ModelChoiceField(
+        queryset=Pokemon.objects.all(),
+        widget=autocomplete.ModelSelect2(
+            url='/battles/pokemon-autocomplete',
+            attrs={
+                'data-placeholder': 'Select the second Pokemon',
+                'data-html': True
+            })
+    )
+
+    third_pokemon = forms.ModelChoiceField(
+        queryset=Pokemon.objects.all(),
+        widget=autocomplete.ModelSelect2(
+            url='/battles/pokemon-autocomplete',
```

Comment:
check if is it possible to use [reverse_lazy](https://docs.djangoproject.com/en/2.0/ref/urlresolvers/#reverse-lazy) here